addons/website_slides/static/src/js/slides_course_fullscreen_player.js

```diff
@@ -182,228 +358,205 @@ odoo.define('website_slides.fullscreen', function (require) {
             }
         },
         /**
+         * Instanciate the quiz widget that will handle all the quiz functionalities
+         *
          * @private
-         * Creates slides objects from every slide-list-cells attributes
          */
-        _getSlides: function (){
+        _renderQuiz: function (){
             var self = this;
-            var slides = $('.o_wslides_fs_sidebar_slide_tab');
-            for (var i = 0; i < slides.length;i++){
-                var slide = $(slides[i]);
-                self.slides.push(slide.data());
-                this._getActiveSlide();
-            }
+            var Quiz = new QuizWidget(this, self.get('slide'), self.nextSlide);
+            Quiz.appendTo('.o_wslides_fs_player');
+            this.$('.next-slide').click(function (){
+                self._goToNextSlide();
+            });
+            this.$('.back-to-video').click(function (){
+                self.goToQuiz = false;
+                self._renderSlideContent();
+            });
         },
         /**
+         * Renders the current slide's content
+         *
          * @private
-         * @param {object} slide
-         * Fetch the quiz for a particular slide
          */
-        _fetchQuiz: function (){
+        _renderSlideContent: function (){
             var self = this;
-            self._rpc({
-                route:"/slide/quiz/get",
-                params: {
-                    'slide_id': self.slide.id
-                }
-            }).then(function (data){
-                if (data){
-                    self.slide.quiz = data;
-                    self._renderPlayer();
-                }
-            });
-        },
-        _fetchHtmlContent: function (){
-            var self = this;
-            self._rpc({
-                route:"/slide/html_content/get",
-                params: {
-                    'slide_id': self.slide.id
-                }
-            }).then(function (data){
-                if (data.html_content) {
-                    self.slide.htmlContent = data.html_content;
-                    self._renderPlayer();
+            var currentSlide = this.get('slide');
+            this.embed_url = $(currentSlide.embed_code).attr('src');
+            if (currentSlide.slide_type === "video"){
+                this.embed_url = "https://" + this.embed_url + "&rel=0&autoplay=1&enablejsapi=1&origin=" + window.location.origin;
+            }
+            var defs = [];
+            defs.push($('.o_wslides_fs_player').html(QWeb.render('website.slides.fullscreen', {
```

Comment:
we should use template rendering only for some slide type (not quiz, not certif, and not video).